queuebacca-core/src/main/java/io/axonif/queuebacca/consumers/ScopedMessageConsumer.java

```diff
@@ -87,14 +93,16 @@ public ScopedMessageConsumer(MessageConsumer<M> consumer, MessageScope messageSc
 	 *     This consumer will run the {@link Message} through a chain of {@link MessageScope MessageScopes} before
 	 *     invoking it's {@link MessageConsumer}.
 	 * </p>
+	 * @return
 	 */
 	@Override
-	public void consume(M message, MessageContext messageContext) {
+	public MessageResponse consume(M message, MessageContext messageContext) {
 		requireNonNull(message);
 		requireNonNull(messageContext);
 
 		ConcreteMessageScopeChain messageScopeChain = new ConcreteMessageScopeChain(new LinkedList<>(messageScopes), message, messageContext);
 		messageScopeChain.next();
+		return null;
```

Comment:
I suppose we should return the MessageResponse from the invocation call` messageScopeChain.next();` instead of just return null here, right?

Reply:
Oops, good catch, that's what it should be doing. Likely code generation from changing the return to void then letting intellij work it's magic.